projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/wpcom-global-styles-view.js

```diff
@@ -68,21 +95,30 @@ document.addEventListener( 'DOMContentLoaded', () => {
 		window.location = upgradeButton.href;
 	} );
 
-	previewButton?.addEventListener( 'click', event => {
+	previewButtonCheckbox?.addEventListener( 'change', event => {
 		event.preventDefault();
-		const checkbox = previewButton.querySelector( 'input[type="checkbox"]' );
-		if ( checkbox ) {
-			checkbox.checked = ! checkbox.checked;
-		}
 		recordEvent( 'wpcom_global_styles_gating_notice_preview', {
-			action: checkbox.checked ? 'show' : 'hide',
+			action: previewButtonCheckbox.checked ? 'show' : 'hide',
 		} );
-		window.location = previewButton.href;
+		window.location = previewButtonCheckbox.dataset.href;
 	} );
 
-	resetButton?.addEventListener( 'click', event => {
+	resetButton?.addEventListener( 'click', async event => {
```

Comment:
Again this might be an issue with my sandboxing but this doesn't seem to be running on the front end? Is there a better way to test these changes?